src/snowflake/cli/plugins/sql/commands.py

```diff
@@ -73,6 +74,11 @@ def execute_sql(
     The command supports variable substitution that happens on client-side. Both &VARIABLE or &{ VARIABLE }
     syntax are supported.
     """
+    exclusive_inputs = {"--filename": files, "--query": query, "--stdin": std_in}
+    incompatible_options = [key for key, value in exclusive_inputs.items() if value]
+    if len(incompatible_options) > 1:
+        raise IncompatibleParametersError(incompatible_options)
+
```

Comment:
Can we handle that on options level?

Reply:
We can, but it will not show the default value for files because typer doesn't support callbacks with type list and default None, there is a discussion in the typer repo https://github.com/tiangolo/typer/discussions/762. 

Instead I added it by this way so it doesn't affect the help option and show the error for incompatible option in sql command.

Reply:
sounds good, thanks!